Mlem/Models/Trackers/Feed/ParentTracker.swift

```diff
@@ -0,0 +1,222 @@
+//
+//  ParentTracker.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2023-10-15.
+//
+
+import Dependencies
+import Foundation
+
+class ParentTracker<Item: TrackerItem>: ObservableObject, ParentTrackerProtocol {
+    @Dependency(\.errorHandler) var errorHandler
+
+    @Published var items: [Item] = .init()
+
+    private var childTrackers: [any ChildTrackerProtocol] = .init()
+    private var internetSpeed: InternetSpeed
+    private var sortType: TrackerSortType
+    
+    var threshold: ContentModelIdentifier?
+    @Published var loadingState: LoadingState = .idle
+
+    init(internetSpeed: InternetSpeed, sortType: TrackerSortType, childTrackers: [any ChildTrackerProtocol]) {
+        self.internetSpeed = internetSpeed
+        self.sortType = sortType
+        self.childTrackers = childTrackers
+
+        for var child in self.childTrackers {
+            child.setParentTracker(self)
+        }
+    }
+
+    func addChildTracker(_ newChild: some ChildTrackerProtocol) {
+        var newChild = newChild
+        newChild.setParentTracker(self)
+    }
+    
+    /// If the given item is the loading threshold item, loads more content
+    /// This should be called as an .onAppear of every item in a feed that should support infinite scrolling
+    func loadIfThreshold(_ item: Item) {
```

Comment:
Yeah, that's a good idea. I've implemented `CoreTracker`, which handles all of the common logic. I also renamed `BasicTracker` to `StandardTracker` because `BasicTracker`/`CoreTracker` does not make it clear which one is the core class and which is the inheriting.